### ISSUE TITLE
Make valid-jsdoc @returns requirement configurable.

### ISSUE THREAD

**Issue by briandipalma** [eslint/eslint#673]
Firstly thanks for the great work on ESLint, very nice project.

I've enabled valid-jsdoc for my code but I'm finding that it's sometimes overly strict.
Specifically the requirement to have an `@returns` specified seems to only make sense when there is a `return` statement in the documented function.

Yet to pass the rule you need to add

`@returns {undefined}`

to many of your JSDoc comments.

Making this explict doesn't seem like a great win for developers or for tools which process the JSDoc. The implicit return value being `undefined` anyway.

I'd like to add another configuration value, `returnRequiredIfReturnUndefined`, to the valid-jsdoc rule.

``` js
"valid-jsdoc": [2, {
    "returnRequiredIfReturnUndefined": false
}]
```

This is how, ideally, it would work.

If there is no return of if the return returns undefined then do not require the developer to fill in a `@return` tag.

Does this seem a sensible change? I can start working on a PR if so.

I have a vague memory of the SpiderMonkey AST having a ReturnNode or something of the type which I could use to achieve this.

I will follow the contributing guidelines.
